### PR TITLE
refactor(api): move log logic into separate package

### PR DESCRIPTION
**Pull request by ecrupper**
Based off of https://github.com/go-vela/server/pull/615 https://github.com/go-vela/server/pull/754 https://github.com/go-vela/server/pull/809 https://github.com/go-vela/server/pull/828 https://github.com/go-vela/server/pull/829 https://github.com/go-vela/server/pull/847 https://github.com/go-vela/server/pull/848 https://github.com/go-vela/server/pull/849 https://github.com/go-vela/server/pull/851 and https://github.com/go-vela/server/pull/853

This change continues the refactor efforts initially introduced in the above PRs.

This adds a new `log` package to the `github.com/go-vela/server/api` package.

This contains all the same handlers that existed previously but with each of them within their own file.

Also changed `GetBuildLogs` to `ListLogsForBuild` to stay consistent.

### CODE REVIEW

File: api/log/list_build.go [hunk 1]
```diff
@@ -0,0 +1,135 @@
...
+
+	// update engine logger with API metadata
+	//
+	// https://pkg.go.dev/github.com/sirupsen/logrus?tab=doc#Entry.WithFields
+	logrus.WithFields(logrus.Fields{
+		"build": b.GetNumber(),
+		"org":   o,
+		"repo":  r.GetName(),
+		"user":  u.GetName(),
+	}).Infof("reading logs for build %s", entry)
```
Comment (cognifloyd) [inline api/log/list_build.go:91]:
```suggestion
	}).Infof("listing logs for build %s", entry)
```

File: api/log/list_build.go [hunk 2]
```diff
@@ -0,0 +1,135 @@
...
+		return
+	}
+
+	// ensure per_page isn't above or below allowed values
+	perPage = util.MaxInt(1, util.MinInt(100, perPage))
+
+	// send API call to capture the list of logs for the build
+	l, t, err := database.FromContext(c).ListLogsForBuild(b, page, perPage)
+	if err != nil {
+		retErr := fmt.Errorf("unable to get logs for build %s: %w", entry, err)
```
Comment (cognifloyd) [inline api/log/list_build.go:118]:
```suggestion
		retErr := fmt.Errorf("unable to list logs for build %s: %w", entry, err)
```

**Review by cognifloyd — CHANGES_REQUESTED** [review]

File: api/log/create_service.go [hunk 3]
```diff
@@ -0,0 +1,131 @@
...
+	"github.com/go-vela/server/router/middleware/org"
+	"github.com/go-vela/server/router/middleware/repo"
+	"github.com/go-vela/server/router/middleware/service"
+	"github.com/go-vela/server/router/middleware/user"
+	"github.com/go-vela/server/util"
+
+	"github.com/go-vela/types/library"
+
+	"github.com/gin-gonic/gin"
+	"github.com/sirupsen/logrus"
```
Comment (cognifloyd) [inline api/log/create_service.go:23]: Could we remove the blank lines between github imports and sort them? Same for the other files.

File: api/log/get_service.go [hunk 4]
```diff
@@ -0,0 +1,98 @@
...
+//   required: true
+//   type: string
+// - in: path
+//   name: build
+//   description: Build number
+//   required: true
+//   type: integer
+// - in: path
+//   name: service
+//   description: ID of the service
```
Comment (cognifloyd) [inline api/log/get_service.go:49]: This is the service number (the number in the build) not the ID (the globally unique service ID) right?

Could we say `Number of the service in the build`? Or more succinctly `Service number` (similar to how we have `Step number` on the step log handlers).

If so, then please also do the same as `GetServiceLog` for `UpdateServiceLog` and `DeleteServiceLog`.

**Review by cognifloyd — APPROVED** [review]

**Review by plyr4 — APPROVED** [review]

**Pull request merged**